### DISCUSSION TITLE
Does Beanie require MongoDB version 4.2 or higher?

### DISCUSSION THREAD

**Discussion by Abashinos** [BeanieODM/beanie#454] [Question]
I've stumbled upon an odd error when attempting to either call `fetch_links` method on a Document instance, or pass the `fetch_links=True` argument to the `find` method on the model itself:
```
pymongo.errors.OperationFailure: Unrecognized pipeline stage name: '$set', full error: {'ok': 0.0, 'errmsg': "Unrecognized pipeline stage name: '$set'", 'code': 40324, 'codeName': 'Location40324'}
```

A short search led me to [this answer](https://stackoverflow.com/a/62625634): 
```
The $set MongoDB Aggregation pipeline stage will work only for MongoDB versions 4.2 and above.

You can use the $addFields Pipeline stage instead.
```

We're using a slightly older version of MongoDB and it might be a hassle to update it (if it would even be possible due to some specific circumstances we're in).

The question is - it possible to backport the `$set` pipeline stage to `$addFields` or is it used in some incompatible way with the older approach?

Btw, a cursory search for a specific MongoDB version requirement in Beanie's PyPI page and docs turned up a definitive answer

Thank you.

**Comment by roman-right** [comment]
Hi! This sounds super reasonable. I'll try my best to make a backport for 4.2

**Reply by Abashinos** [reply]
> Hi! This sounds super reasonable. I'll try my best to make a backport for 4.2

Hi @roman-right!
Thank you for the help!
Actually, we need a backport for 4.0 since 4.2 is the earliest version to support the new `$set` pipeline operation.

I've done some research on a somewhat related issue and [commented](https://github.com/roman-right/beanie/issues/250#issuecomment-1362509658) my findings, which are basically as follows:
Passing of `fetch_links=True` argument can cause a validation issue in MongoDB **4.2** due to unsupported `$` as the starting symbol of variable names, but even if that one is fixed, it would still a syntax error in MongoDB **4.0** due to unsupported `$set` pipeline operation.

It seems to me that fixing the linked [issue](https://github.com/roman-right/beanie/issues/250) **and** replacing `$set` operation with `$addFields` may resolve both problems we're facing, but, apart from that, they're unrelated.

Hope that helps.
Thanks again!

**Reply by roman-right** [reply]
Thank you! This will help :)

**Reply by Abashinos** [reply]
Hi @roman-right!
Is there any ETA on this backport?
Thank you!

**Comment by Abashinos** [comment]
bump
